esp/services/esdl_svc_engine/esdl_store.hpp

```diff
@@ -0,0 +1,82 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2017 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef _ESDL_STORE_HPP__
+#define _ESDL_STORE_HPP__
+
+#include "jlib.hpp"
+#include "jstring.hpp"
+#include "jptree.hpp"
+
+#ifdef ESDL_ENGINE_EXPORTS
+ #define esdl_engine_decl DECL_EXPORT
+#else
+ #define esdl_engine_decl
+#endif
+
+interface IEsdlStore : public IInterface
+{
+    virtual void fetchDefinition(const char* id, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinition(const char* esdlServiceName, unsigned targetVersion, StringBuffer& esxdl) = 0;
+    virtual void fetchDefinitionByNameOnly(const char * name, StringBuffer & def) = 0;
+    virtual IPropertyTree* fetchBinding(const char* espProcess, const char* bindingName) = 0;
+    virtual bool isDefinitionBound(const char * esdldefid) = 0;
+    virtual bool isDefinitionBound(const char * esdldefname, int version) = 0;
+    virtual IPropertyTree* getDefinitionRegistry(const char * wsEclId, bool readonly) = 0;
```

Comment:
wsEclId is not a good name... but maybe that parameter is not even used?  Returning the registry as a property tree isn't a good idea... when we move this stuff from DALI it may have to change.  It works for DALI because DALI is already like a giant remote property tree.

Reply:
Removed argument wsEclId as it's not being used in the function.

Reply:
I think we can keep the return type as IPropertyTree for now, being aware that this may need to change.